crates/oxc_wasm/src/lib.rs

```diff
@@ -331,13 +331,21 @@ impl Oxc {
         ) {
             let space = " ".repeat(depth * 2);
 
+            let scope_tree = semantic.scopes();
             for scope_id in scope_ids {
-                let flags = semantic.scopes().get_flags(*scope_id);
-                let next_scope_ids = semantic.scopes().get_child_ids(*scope_id);
+                let flags = scope_tree.get_flags(*scope_id);
+                let child_scope_ids = scope_tree
+                    .descendants_from_root()
+                    .filter(|id| {
+                        scope_tree
+                            .get_parent_id(*id)
+                            .is_some_and(|parent_id| parent_id == *scope_id)
+                    })
+                    .collect::<Vec<_>>();
```

Comment:
Using a linear scan, performance shouldn't be that bad ...

Reply:
I think this is O($n^2$).`write_scope_text` is called for every scope, and it iterates over all scopes again in `let child_scope_ids = ...` (though it does nothing for most of them).

If we can assume that scopes are in traversal order (children directly after their parent), I think it's doable in O(n). That assumption would hold true if `ScopeTree` has come straight out of semantic, but it wouldn't if it's been through transformer. Do we want to support displaying the tree post-transform?

But does it matter anyway? If I understand right, this is just for playground where typically people won't paste in huge tracts of code with loads of scopes.